crates/sui-bridge-indexer/src/main.rs

```diff
@@ -86,18 +91,26 @@ async fn main() -> Result<()> {
         .map_err(|e| anyhow::anyhow!(e.to_string()))?,
     );
 
-    let unfinalized_handle = eth_worker.start_indexing_unfinalized_events(eth_client.clone());
-    let finalized_handle = eth_worker.start_indexing_finalized_events(eth_client.clone());
-
+    let unfinalized_handle = eth_worker.start_indexing_unfinalized_events(eth_client.clone()).await.unwrap();
+    let finalized_handle = eth_worker.start_indexing_finalized_events(eth_client.clone()).await.unwrap();
+    let handles = vec![unfinalized_handle, finalized_handle];
     // TODO: add retry_with_max_elapsed_time
-    let progress = start_processing_sui_checkpoints(
-        &config_clone,
-        db_url,
-        indexer_meterics,
-        ingestion_metrics,
-    );
+    if let Some(sui_rpc_url) = config.sui_rpc_url.clone() {
```

Comment:
I'd like to talk about how we plug in a data reader. Let's make sure to talk at our daily later